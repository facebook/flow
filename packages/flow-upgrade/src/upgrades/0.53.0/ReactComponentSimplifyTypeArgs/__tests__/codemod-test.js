/**
 * @format
 */

'use strict';

const path = require('path');
const fs = require('fs');
const cp = require('child_process');
const jscodeshiftFlowParser = require('jscodeshift/parser/flow');
const jscodeshift = require('jscodeshift').withParser(jscodeshiftFlowParser);
const transform = require('../codemod');

const FIXTURES_DIR = path.join(__dirname, './fixtures');

fs.readdirSync(FIXTURES_DIR).forEach(fixture => {
  test(fixture, () => {
    const file = path.join(FIXTURES_DIR, fixture);
    const source = fs.readFileSync(file, 'utf8');
    const root = jscodeshift(source);
    const skipped = !transform(jscodeshift, root);
    expect(skipped ? null : root.toSource()).toMatchSnapshot();
  });
});
