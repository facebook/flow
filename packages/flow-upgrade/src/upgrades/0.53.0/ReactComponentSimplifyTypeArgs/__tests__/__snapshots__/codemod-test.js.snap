// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All.js 1`] = `
"// @flow

import React from 'react';

class MyComponent extends React.Component<Props, State> {
  static defaultProps: DefaultProps = {};
  state: State = {};

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props, State> {
    static defaultProps: DefaultProps = {};
    state: State = {};

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`ComponentNamedImport.js 1`] = `
"// @flow

import {Component} from 'react';

class MyComponent extends Component<Props, State> {
  static defaultProps: DefaultProps = {};
  state: State = {};

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends Component<Props, State> {
    static defaultProps: DefaultProps = {};
    state: State = {};

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`DefaultProps.js 1`] = `
"// @flow

import React from 'react';

class MyComponent extends React.Component<Props> {
  static defaultProps: DefaultProps = {};

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props> {
    static defaultProps: DefaultProps = {};

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`DefaultPropsInferred.js 1`] = `
"// @flow

import React from 'react';

class MyComponent extends React.Component<Props> {
  static defaultProps = {};

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props> {
    static defaultProps = {};

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`DefaultPropsInline.js 1`] = `
"// @flow

import React from 'react';

class MyComponent extends React.Component<Props> {
  static defaultProps: {a: number, b: number, c: number} = {a: 1, b: 2, c: 3};

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props> {
    static defaultProps: {a: number, b: number, c: number} = {a: 1, b: 2, c: 3};

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`DefaultPropsNone.js 1`] = `
"// @flow

import React from 'react';

class MyComponent extends React.Component<Props> {
  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props> {
    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`DefaultPropsNoneButState.js 1`] = `
"// @flow

import React from 'react';

class MyComponent extends React.Component<Props, State> {
  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props, State> {
    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`DefaultPropsTypeArgNoProperty.js 1`] = `
"// @flow

import React from 'react';

class MyComponent extends React.Component<Props, State> {
  static defaultProps: any;
  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props, State> {
    static defaultProps: any;
    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`DefaultPropsUninitialized.js 1`] = `
"// @flow

import React from 'react';

class MyComponent extends React.Component<Props> {
  static defaultProps: DefaultProps;

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props> {
    static defaultProps: DefaultProps;

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`FlowNone.js 1`] = `
"import React from 'react';

class MyComponent extends React.Component<Props, State> {
  static defaultProps: DefaultProps = {};
  state: State = {};

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props, State> {
    static defaultProps: DefaultProps = {};
    state: State = {};

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`PureComponent.js 1`] = `
"// @flow

import React from 'react';

class MyComponent extends React.PureComponent<Props, State> {
  static defaultProps: DefaultProps = {};
  state: State = {};

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.PureComponent<Props, State> {
    static defaultProps: DefaultProps = {};
    state: State = {};

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`PureComponentNamedImport.js 1`] = `
"// @flow

import {PureComponent} from 'react';

class MyComponent extends PureComponent<Props, State> {
  static defaultProps: DefaultProps = {};
  state: State = {};

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends PureComponent<Props, State> {
    static defaultProps: DefaultProps = {};
    state: State = {};

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;

exports[`ReactNone.js 1`] = `null`;

exports[`ReactRequire.js 1`] = `
"// @flow

const React = require('react');

class MyComponent extends React.Component<Props, State> {
  static defaultProps: DefaultProps = {};
  state: State = {};

  defaultProps: T;
  static props: T;
  static state: T;
  a: T;
  b = 5;
  c: T = 5;
  method() {}
}

const expression = () =>
  class extends React.Component<Props, State> {
    static defaultProps: DefaultProps = {};
    state: State = {};

    defaultProps: T;
    static props: T;
    static state: T;
    a: T;
    b = 5;
    c: T = 5;
    method() {}
  }
"
`;
